.github/workflows/lockfiles.yaml

```diff
@@ -0,0 +1,594 @@
+                --output "${STDERR_LOG}.${theme}.${ext}" -
+            done
+          done
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v5.0.0
+        with:
+          aws-region: ${{ env.AWS_REGION }}
+          role-session-name: GitHubActions-${{ github.run_id }}
+          role-to-assume: arn:aws:iam::${{ env.AWS_ACCOUNT_ID }}:role/GitHubActions@org=${{ github.repository_owner }},repo=${{ github.event.repository.name }}
+          allowed-account-ids: ${{ env.AWS_ACCOUNT_ID }}
+          # This should be the minimal set of permissions needed by this workflow.
+          inline-session-policy: >-
+            {
+              "Version": "2012-10-17",
+              "Statement": [
+                {
+                  "Sid": "AllowPutObject",
+                  "Effect": "Allow",
+                  "Action": ["s3:PutObject", "s3:PutObjectTagging"],
+                  "Resource": ["arn:aws:s3:::${{ env.AWS_S3_BUCKET }}/${{ env.AWS_S3_BUCKET_ROOT }}/*"]
+                }
+              ]
+            }
+
+      - name: Upload lockfile diff picture to S3
+        # GitHub flavored markdown does not accept data:image/png URIs for images.
+        # And GitHub does not provide any convenient APIs for uploading image assets.
+        # So, we upload to AWS and save the URI for in generated markdown below.
+        if: steps.lockfile.outputs.CHANGED == 'true'
+        # NOTE: The packaged s3 actions are too old, incomplete, or do not allow setting content-type + tags.
+        env:
+          AWS_S3_PATH: ${{ env.AWS_S3_BUCKET_ROOT }}/lockfile-${{ matrix.resolve }}
+        run: |
+          aws s3 cp \
+            "${STDERR_LOG}.github-dark.png" \
+            "s3://${{ env.AWS_S3_BUCKET }}/${{ env.AWS_S3_PATH }}/" \
+            --content-type image/png
+          aws s3api put-object-tagging \
+            --expected-bucket-owner "${AWS_ACCOUNT_ID}" \
+            --bucket "${{ env.AWS_S3_BUCKET }}" \
+            --key "${{ env.AWS_S3_PATH }}/$(basename ${STDERR_LOG}.github-dark.png)" \
+            --tagging '{"TagSet":[{"Key":"Service","Value":"GitHub Actions"}]}'
+
+      - name: Prepare text-only lockfile diff for commit message
+        if: steps.lockfile.outputs.CHANGED == 'true'
+        env:
+          # This sed script replaces ansi escape chars with unicode for use in the commit msg.
+          SED_SCRIPT: |
+            /^\x1B\[4m\( \+\)\(\x1B\[0m\)\?$/{ # line above headings (spaces with ansi underline)
+              s/ /_/g; # use underline char instead of space
+            }
+            # the next one adds a line of overline chars to replace the ansi overline
+            /^== .* ==$/{ # heading text line (the matched line goes in pattern space)
+              h;       # save copy of heading line in "hold" space
+              s/./‾/g; # make an line of overline chars to replace the heading's ansi underline
```

Comment:
The `‾` (unicode "overline" character) is visible in the github UI when I'm using a desktop browser. And it shows up nicely in `git log` output on my linux machine. On my android phone, it shows up as a box, so the fonts apparently don't have that character. A line of boxes still basically underlines the previous header line, so `‾` was the best I could do with raw unicode chars.